core/arch/aarch64/emit_utils.c

```diff
@@ -605,8 +604,7 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
                 INSTR_CREATE_ldr(
                     dcontext, opnd_create_reg(DR_REG_P0 + i),
                     opnd_create_base_disp(
-                        DR_REG_X1, DR_REG_NULL, 0,
-                        i * (proc_get_vector_length_bytes() / 8),
+                        DR_REG_X1, DR_REG_NULL, 0, i * sizeof(dr_simd_t),
```

Comment:
Wait, are the pred regs stored in a dr_simd_t?  Indeed they are in mcxt_api.h: but aren't they 1/8 the size?  Isn't that a waste of space?  @AssadHashmi could you help us understand that?

Reply:
@jackgallagher-arm any thoughts on this?

Reply:
Yes, predicates and FFR are 1/8th of the size. The idea was to get SVE working first then make such refinements, and this was a quick way of getting things up and running. Should we raise an issue to fix this or do you want to do it as part of this patch?

Reply:
Separate issue would be better, since this PR is needed to unblock our downstream merge.

Reply:
Created https://github.com/DynamoRIO/dynamorio/issues/6760.

Reply:
Thanks!

Reply:
Note that such a change in the mcontext layout would break compatibility and require a major version change.

Reply:
Oh you mean PR #6725 changes the size of dr_simd_t!  That didn't sink in when I reviewed it.  Yes, it breaks compatibility and it should have bumped the minor version to 90 as part of the same PR per https://dynamorio.org/page_new_release.html#autotoc_md230 and clearly documented the breakage in the release notes.  @AssadHashmi please submit a follow-up PR for that.

Reply:
Wait, PR #6725 doesn't change the size: the new union field member is the same 64 bytes.

@abhinav92003 please clarify what you mean about breaking compatibility

Reply:
PR #6725 doesn't break compatibility. But it sets the data differently in the existing field. I assume that also needs to be backward compatible (not just the struct definition).

Reply:
Changing how we set data is set in the existing same-size field also would require clients to be updated. They would need to read the struct accordingly.

Reply:
Before, the SIMD data was written in a way that didn't even line up with the dr_simd_t structs, right?  Someone trying to manually pull the data out would be very confused.  In fact: was there a users list question about that with someone not able to get the data?  This seems like a bug fix.  Maybe someone figured out the way it was written before and wrote client code assuming that but it seems like it's not the same kind of change where we broke a documented interface.

Reply:
Right. That's what I meant.

> Though it really fixes it by spilling to the slots one would expect